Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift

```diff
@@ -110,19 +110,9 @@ extension BrowserViewController: URLBarDelegate {
         }
     }
 
-    private func presentFakespotViewController() {
-        let fakespotViewController = FakespotViewController()
-        if #available(iOS 15.0, *) {
-            if let sheet = fakespotViewController.sheetPresentationController {
-                sheet.detents = [.medium(), .large()]
-            }
-        }
-        present(fakespotViewController, animated: true)
-    }
-
     func urlBarDidPressShoppingCart(_ urlBar: URLBarView, shoppingCart: UIButton) {
         TelemetryWrapper.recordEvent(category: .action, method: .tap, object: .shoppingCartButton)
-        presentFakespotViewController()
+        navigationHandler?.showFakespotFlow()
```

Comment:
Could it happen that a user clicks that button twice very rapidly? If so I think two coordinators could be spawn?

Reply:
Yes, in this case we need would need some throttling mechanism for tapping a button or a check to see if it's already presented.

Reply:
In some other places we have something like this:
```
        guard !childCoordinators.contains(where: { $0 is SettingsCoordinator}) else {
            return false // route is handled with existing child coordinator
        }
```
So we pass the responsibility to the child coordinator. Since we don't have any `handle(route:)` implemented in that new fakeSpotCoordinator it would do nothing (since it would already be showing). So I think this would work. If we use the same solution everywhere we could have something more reusable/general at one point so this is easier to support this kind of edge cases